scripts/modules/b2b-account/account-address-search.js

```diff
@@ -0,0 +1,270 @@
+    "underscore",
+    "hyprlive",
+    "modules/backbone-mozu",
+    "hyprlivecontext",     
+    "modules/models-customer",
+    'modules/mozu-grid/mozugrid-view',
+    'modules/mozu-grid/mozugrid-pagedCollection',   
+    "modules/views-paging",
+    'modules/editable-view',
+    "modules/models-quotes",
+    "modules/models-b2bcontacts"], 
+    function ($, api, _, Hypr, Backbone, HyprLiveContext, CustomerModels, MozuGrid, MozuGridCollection,
+    PagingViews, EditableView, QuoteModels, B2bModels) {
+        var timeComponent = "T00:00:00z";
+        var filterstring = "";
+        var timeout = null;
+        var pubsub = {};
+       
+       
+    var B2bContactsMozuGrid = MozuGrid.extend({
+        render: function () {
+            var self = this;
+            MozuGrid.prototype.render.apply(self, arguments);
+        }
+    });
+
+    var B2bContactsView = Backbone.MozuView.extend({
+        templateName: "modules/b2b-account/account-address-search/account-address-search",
+        initialize: function () {       
+          Backbone.MozuView.prototype.initialize.apply(this, arguments);
+         
+            
+        },        
+        render: function () {
+            var self = this;
+            Backbone.MozuView.prototype.render.apply(this, arguments);
+            var collection = new B2bContactsGridCollectionModel({ autoload: true }); 
+            _.extend(collection, Backbone.Events);
+            collection.bind('custom:event',this.callbackForGridSelection, this);            
+            this.initializeGrid(collection);     
+            this.$modal = this.$('.modal');
+        },      
+        initializeGrid: function (collection) {
+            var self = this;
+            self._b2bContactGridView = new B2bContactsMozuGrid({
+                el: $('.mz-b2b-contacts-grid'),
+                model: collection
+            });
+            $('[data-mz-action="addAddressfilter"]').on('keyup input', function(e) {
+                e.preventDefault();                                                
+                clearTimeout(timeout);        
+                if(($("#searchAddress").val().length>=3) || ($("#searchAddress").val().length === 0))
+                {
+                timeout = setTimeout(function () {
+                    self.filterGrid(collection, self);
+                }, 400); 
+            }
+            }); 
+        
+            $('[data-mz-action="addCityfilter"]').on('keyup input', function(e) {
+                e.preventDefault();                                                
+                clearTimeout(timeout); 
+                if(($("#searchCity").val().length>=3) || ($("#searchCity").val().length === 0))
+                {            
+                timeout = setTimeout(function () {
+                    self.filterGrid(collection, self);
+                }, 400); 
+            }
+            });  
+            $('[data-mz-action="addStatefilter"]').on('keyup input', function(e) {
```

Comment:
Convert this to a common function. So, we can reuse this something like

functionName: function(mzDataAction, ){
           $('[data-mz-action="'+mzDataAction+'"]').on('keyup input', function(e) {
                e.preventDefault();                                                
                clearTimeout(timeout);        
                if(($("#searchState").val().length>=3) || ($("#searchState").val().length === 0))
                {     
                timeout = setTimeout(function () {
                    self.filterGrid(collection, self);
                }, 400); 
            }
            }); 
}

pass the necessary information to function as params like mzDataAction and get rid of duplicate code.